VideoLocker/src/main/java/org/edx/mobile/module/notification/NotificationDelegate.java

```diff
@@ -0,0 +1,39 @@
+package org.edx.mobile.module.notification;
+
+import org.edx.mobile.model.api.EnrolledCoursesResponse;
+import java.util.List;
+
+/**
+ *  abstracted for different implementation.
+ *
+ */
+public interface NotificationDelegate {
+    public void unsubscribeAll( );
+
+    /**
+     * if user login,  he should subscribe channels based on local pref setting.
+     */
+    public void resubscribeAll();
+
+    //check if local subscribed is not in parse server
+    //then try to subscribe it
+    public void syncWithParse();
```

Comment:
@aleffert  it may not make much sense for abstracting notification service. because different notification  middleware may invoke different workflow. 
But i is always a good idea to program on interface.

Just renamed.